heimlig/src/hsm/workers/ecc_worker.rs

```diff
@@ -74,8 +79,8 @@ impl<
                 error: Error::KeyStore(e),
             },
             Ok(key_info) => {
-                let key_exists = locked_key_store.deref().is_stored(key_id);
-                if key_exists && !key_info.permissions.overwrite {
+                let key_exists = locked_key_store.deref().is_key_available(key_id);
+                if key_exists && (!overwrite || !key_info.permissions.overwrite) {
```

Comment:
Does it make sense to rely only on the check from `import_symmetric_key` and `import_key_pair`?